Lesson1HomeTask/task2/main.cpp

```diff
@@ -0,0 +1,80 @@
+#include <iostream>
+#include <cstdint>
+#include <list>
+#include <cmath>
+#include <cstdlib>
+
+uint64_t make_specific_number (uint32_t base, uint32_t power)
+{
+    std::list<uint64_t> values;
+
+    for (uint32_t i = 1; i <= power; i++ )
+        values.push_back (std::pow(base, i));
+
+    if (values.empty())
+        return 0;
+
+    uint64_t result = 0;
+
+    for (auto& el : values)
+    {
+        uint8_t last_num = el % 10;
+        uint64_t first_num = el;
+
+        while (first_num >= 10)
+            first_num /= 10;
+
+        if (0 != result)
+            result *= 100;
+        result += first_num*10 + last_num;
+    }
+
+    return result;
+}
+
+uint64_t make_specific_number_2 (uint32_t base, uint32_t power)
+{
+    std::list<std::string> values;
+
+    for (uint32_t i = 1; i <= power; i++ )
+        values.push_back (std::to_string (static_cast<uint64_t> (std::pow (base, i))));
```

Comment:
What the point of using concrete types like uint64, uint32 here? Why size_t is not enough?